cmake/multi_image.cmake

```diff
@@ -173,7 +173,130 @@ function(get_child_image_cmake_args child_image output_cmake_args)
 
   unset(${output_cmake_args} PARENT_SCOPE)
   set(${output_cmake_args} "${child_image_cmake_args}" PARENT_SCOPE)
+endfunction()
 
+# Adds an S1 variant of an image to the build.
+#
+# Required arguments are:
+# SOURCE_DIR - The source dir of the base image.
+#
+# Optional argumnets are:
+# BASE_IMAGE - The name of the base image which the variant image is based on.
+function(add_s1_variant)
+  set(oneValueArgs BASE_IMAGE SOURCE_DIR)
+  cmake_parse_arguments(ASV "" "${oneValueArgs}" "" ${ARGN})
+
+  if (NOT ASV_SOURCE_DIR)
+    message(FATAL_ERROR "Missing parameter, required: SOURCE_DIR")
+  endif()
+
+  set(s1_override_conf
+    ${ZEPHYR_NRF_MODULE_DIR}/subsys/bootloader/image/is_s1_image.conf)
```

Comment:
This makes much more sense. It also simplifies the process of appending the extra config since we no longer need to search-replace the OVERLAY_CONFIG.